src/globals.js

```diff
@@ -13,6 +13,22 @@ export const SpansContainer = (() => {
   return { addSpan, getSpans, clearSpans };
 })();
 
+export const GlobalTimer = (() => {
+  let currentTimer = undefined;
+
+  const setGlobalTimeout = (func, duration) => {
+    clearTimer();
+    currentTimer = setTimeout(func, duration);
+    currentTimer.unref();
```

Comment:
so it might never be called right? if the lambda exists somehow before the timeout

Reply:
Yes, to protect us from the `callbackWaitsForEmptyEventLoop`

Reply:
Yes: "If there is a Timeout object scheduled using a set function, unref() can be called on that object. This will change the behavior slightly, and not call the Timeout object if it is the last code to execute. The Timeout object will not keep the process alive, waiting to execute."
https://nodejs.org/uk/docs/guides/timers-in-node/

Reply:
Yes, Nir you right
This is the exact behavior we want